### PR TITLE
Update releasing instructions

### PR DESCRIPTION
**Pull request by rebello95**
Update the instructions to include step-by-step, as well as mentioning Carthage project re-generation.

### CODE REVIEW

**Comment by rebello95** [comment]
Re-running CI

File: README.md [hunk 1]
```diff
@@ -170,11 +170,16 @@ Please get involved! See our [guidelines for contributing](CONTRIBUTING.md).
...
+    `$ pod spec lint SwiftGRPC.podspec`
+    
+1. Update the Carthage Xcode project (diff will need to be checked in with the version bump):
 
-    $ pod trunk push SwiftGRPC.podspec
+    `$ make project-carthage`
+    
+1. Bump the version in the `SwiftGRPC.podspec` file
+
+1. Merge these changes, then create a new `Release` with corresponding `Tag`. Be sure to include a list of changes in the message
```
Comment (MrMage) [inline README.md]: Should we also add the `pod spec push` command as the last step?

Comment (rebello95) [reply README.md]: Added!

**Comment by MrMage** [comment]
Re-running CI... I guess we really need to look into these flakes soon.

**Pull request merged**